app/lib/actions/katello/repository/destroy.rb

```diff
@@ -16,6 +16,7 @@ def plan(repository, options = {})
           plan_action(Pulp::Repository::Destroy, pulp_id: repository.pulp_id)
           plan_action(Product::ContentDestroy, repository) unless skip_environment_update
           plan_action(ElasticSearch::Repository::Destroy, pulp_id: repository.pulp_id)
+          plan_action(Katello::Repository::DestroyMedia, repository)
```

Comment:
Does this need to be higher in the order to ensure it fails early enough if it is going to fail?